src/services/policy.service.ts

```diff
@@ -52,4 +53,84 @@ export class PolicyService implements PolicyServiceAbstraction {
         await this.storageService.remove(Keys.policiesPrefix + userId);
         this.policyCache = null;
     }
+
+    async getMasterPasswordPolicyOptions(): Promise<MasterPasswordPolicyOptions> {
+        const policies: Policy[] = await this.getAll(PolicyType.MasterPassword);
+        let enforcedOptions: MasterPasswordPolicyOptions = null;
+
+        if (policies == null || policies.length === 0) {
+            return enforcedOptions;
+        }
+
+        policies.forEach((currentPolicy) => {
+            if (!currentPolicy.enabled || currentPolicy.data == null) {
+                return;
+            }
+
+            if (enforcedOptions == null) {
+                enforcedOptions = new MasterPasswordPolicyOptions();
+            }
+
+            if (currentPolicy.data.minComplexity != null
+                && currentPolicy.data.minComplexity > enforcedOptions.minComplexity) {
+                enforcedOptions.minComplexity = currentPolicy.data.minComplexity;
+            }
+
+            if (currentPolicy.data.minLength != null
+                && currentPolicy.data.minLength > enforcedOptions.minLength) {
+                enforcedOptions.minLength = currentPolicy.data.minLength;
+            }
+
+            if (currentPolicy.data.requireUpper) {
+                enforcedOptions.requireUpper = true;
+            }
+
+            if (currentPolicy.data.requireLower) {
+                enforcedOptions.requireLower = true;
+            }
+
+            if (currentPolicy.data.requireNumbers) {
+                enforcedOptions.requireNumbers = true;
+            }
+
+            if (currentPolicy.data.requireSpecial) {
+                enforcedOptions.requireSpecial = true;
+            }
+        });
+
+        return enforcedOptions;
+    }
+
+    async evaluateMasterPassword(passwordStrength: number, newPassword: string,
```

Comment:
I don't see any need for async here?

Reply:
I believe you're correct. Fixing on updated PR.